examples/jsm/webxr/ARButton.js

```diff
@@ -1,40 +1,38 @@
-var ARButton = {
+class ARButton {
 
-	createButton: function ( renderer, sessionInit = {} ) {
+	static createButton( renderer, sessionInit = {} ) {
 
-		function showStartAR( /*device*/ ) {
+		let currentSession = null;
 
-			var currentSession = null;
+		const button = document.createElement( 'button' );
+		const message = document.createElement( 'a' );
 
-			function onSessionStarted( session ) {
+		function onSessionEnded( /*event*/ ) {
 
-				session.addEventListener( 'end', onSessionEnded );
+			if ( currentSession )
+				currentSession.removeEventListener( 'end', onSessionEnded );
 
-				/*
-				session.updateWorldTrackingState( {
-					'planeDetectionState': { 'enabled': true }
-				} );
-				*/
+			button.textContent = 'START AR';
 
-				renderer.xr.setReferenceSpaceType( 'local' );
-				renderer.xr.setSession( session );
-				button.textContent = 'STOP AR';
+			currentSession = null;
 
-				currentSession = session;
+		}
 
-			}
+		function onSessionStarted( session ) {
 
-			function onSessionEnded( /*event*/ ) {
+			session.addEventListener( 'end', onSessionEnded );
 
-				currentSession.removeEventListener( 'end', onSessionEnded );
+			renderer.xr.setReferenceSpaceType( 'local' );
+			renderer.xr.setSession( session );
+			button.textContent = 'STOP AR';
 
-				button.textContent = 'START AR';
+			currentSession = session;
 
-				currentSession = null;
+		}
 
-			}
+		function showStartAR( /*device*/ ) {
 
-			//
+			this.currentSession = null;
```

Comment:
You have a `let currentSession` and `this.currentSession`.

Reply:
good catch, will change